### DISCUSSION TITLE
Geo fences support

### DISCUSSION THREAD

**Discussion by predictron-cloud** [HaylLtd/cordova-plugin-background-geolocation#46] [Q&A]
Hi guys!

I see "circular region monitoring" in the description of the plugin. Is it something similar to geo fences? Is there any opportunity to implement geo fences functionality (with 1000+ fences) using this plugin? If custom codding should be done, where is the most relevant place in code to put it for android?

**Comment by HarelM** [comment]
Thanks for reaching out! :-)
Both @raddishiow and I have forked this project in order to be able to maintain it, but we didn't write it from scratch nor experts in it.
The best approach would be to research the code, and submit a pull request.
But before you do so I would be happy if you could explain the motivation behind this feature from a UX point of view, and from a developer point of view.
Thanks!

**Reply by predictron-cloud** [reply]
From UX point of view, it could be something like reminder notification at some places, which user saved in Application. There are some apps for it in AppStore. Also there is a plugin for it https://www.npmjs.com/package/floydspace-cordova-plugin-geofence. But this plugin supports only 200 geofences for android. State-of-the-art plugin from transistorsoft can do it, but I need open source solution, as I'm developing open source project.
From developer's point of view: there should be some SQLite database with places and when user is moving, background job should retrieve the nearest points and add them as standard Android Geo Fences.

**Reply by HarelM** [reply]
I guess you have two options then - 
1. fork/send PR to geofence plugin
2. fork/send PR to this plugin

From what I gather, the other plugin seems like a more suitable location for this improvement. But that maybe just me...
